core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -45,18 +46,23 @@ public DataExpressionSource(final ValueExpression dataExpression, final int inde
     }
 
     @Override
-    protected byte[] getData(final long offset, final int size) throws IOException {
+    protected byte[] getData(final long offset, final BigInteger length) throws IOException {
+        if (!isAvailable(offset, length)) { throw new IOException("Data to read is not available."); }
```

Comment:
You may consider extending the error messages, to differentiate between the different sources. This difference could be useful for debugging, whether the problem is with a tie or the complete format.